libraries/AP_BoardConfig/AP_BoardConfig.cpp

```diff
@@ -418,7 +428,26 @@ void AP_BoardConfig::init()
         printf("SDCard failed to start\n");
     }
 #endif
+
+#if AP_CRASHDUMP_ACK_ENABLED
+    crashdump_ack.check_reset();
+#endif
+}
+
+#if AP_CRASHDUMP_ACK_ENABLED
+void AP_BoardConfig::CrashDump::check_reset()
+{
+    // if there is no crash_dump.bin then clear the crash dump ack.
+    // This means on subsequent crash-dumps appearing the user must
+    // re-acknowledge.
+    int fd = AP::FS().open("@SYS/crash_dump.bin", O_RDONLY);
```

Comment:
better to use hal.util->last_crash_dump_size() > 0

Reply:
Done